### DISCUSSION TITLE
Lag when moving from one tab with tree to other tab with tree

### DISCUSSION THREAD

**Discussion by cristinathoughtpennies** [AvaloniaUI/Avalonia#10258] [Q&A]
Hello!
What I am experiencing: I have 2 tabs with TreeViews in them. The tab I am currently on has the tree completely expanded, and has around 80 nodes. The other tab has 1 node. If I try to move from the current tab to the other tab, the app freezes for an unknown amount of time and seems to be looping through NotifyResourcesChanged.
Is there something that I am missing or did for my app to lag this badly? I saw this discussion: https://github.com/AvaloniaUI/Avalonia/discussions/6365 but i am nowhere near their amount of nodes, so I wouldn't think it would be the same issue.
Any help would be much appreciated!

**Comment by maxkatz6** [comment]
Normally, https://github.com/AvaloniaUI/Avalonia/discussions/6365 is an answer.
Hard to be more helpful here without seeing the problem and code.

**Reply by cristinathoughtpennies** [reply]
Progress! Ive been looking through my source control history, and the lag started after I updated packages from `11.0.999-cibuild0028475-beta` to `11.0.999-cibuild0028887-beta`, and it seems that the first package to have it start happening was `11.0.999-cibuild0028768-beta`

**Reply by cristinathoughtpennies** [reply]
Is there a way to see what changes went into this build?

**Reply by maxkatz6** [reply]
This one seems like https://github.com/AvaloniaUI/Avalonia/pull/10004
Shouldn't really affect though.
![image](https://user-images.githubusercontent.com/3163374/218030470-483a2b4d-ee61-4b1f-861c-455b7fce89b2.png)

**Comment by timunie** [comment]
@cristinathoughtpennies are you able to send a minimal sample showing your issue?

**Reply by MrJul** [reply]
Not the OP, but I've often had the dev tools open recently and noticed this behavior several times.
Here's a quick repro, latest master:

1. Run ControlCatalog.NetCore
2. Open the dev tools
3. Switch to the Visual Tree tab
4. Select one of the colored square from the catalog using Ctrl+Shift
5. Switch to the Logical Tree tab
6. The app should be frozen with a call stack similar to:

![image](https://user-images.githubusercontent.com/1623034/218078801-6aef92eb-da10-4154-885d-48d6018752ef.png)

Doesn't seem to stack overflow though.

**Reply by cristinathoughtpennies** [reply]
Thank you @MrJul! Yes this is what ive been experiencing

**Reply by maxkatz6** [reply]
It is expected to invalidate resources for the tree view nodes, as they are added and removed from the tree of elements (as TabControl doesn't keep alive all tabs at once, but only one of them).
Can somebody check with dotTrace maybe?

**Reply by MrJul** [reply]
See #10413 for a (bit longer) description of the problem encountered here.
